mlflow/projects/__init__.py

```diff
@@ -135,12 +136,25 @@ def run(uri, entry_point="main", version=None, parameters=None, experiment_id=No
     :param run_id: Note: this argument is used internally by the MLflow project APIs and should
                    not be specified. If specified, the run ID will be used instead of
                    creating a new run.
+    :param cluster_spec_dict: When ``mode`` is "databricks", dictionary corresponding to a
+                              `Databricks cluster specification
+                              <https://docs.databricks.com/api/latest/jobs.html#clusterspec>`_
+                              to use when launching a run. If specified, takes precedence over
+                              ``cluster_spec``.
```

Comment:
As above, can we just reuse the cluster_spec parameter? If it's a string, treat it as a file name, and if it's a dict, treat it as a JSON. It's confusing to have two parameters for this, especially in a dynamically typed language like Python.